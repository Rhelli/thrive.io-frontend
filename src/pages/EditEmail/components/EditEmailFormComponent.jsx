import React from 'react';
import PropTypes from 'prop-types';
import styles from './EditEmailFormComponent.module.scss';

const EditEmailFormComponent = ({ userProfile }) => {
  const { email } = userProfile;

  return (
    <div className={styles.editEmailFormContainer}>
      <form>
        <div className={styles.editEmailRow}>
          <label htmlFor="email">
            <h3>Your Email Address</h3>
            <input id="newEmail" type="email" placeholder={email} />
          </label>
        </div>
        <div className={styles.editEmailSubmitButton}>
          <button type="submit">
            Change Email
          </button>
        </div>
      </form>
    </div>
  );
};

EditEmailFormComponent.propTypes = {
  userProfile: PropTypes.shape({
    email: PropTypes.string.isRequired,
  }).isRequired,
};

export default EditEmailFormComponent;
