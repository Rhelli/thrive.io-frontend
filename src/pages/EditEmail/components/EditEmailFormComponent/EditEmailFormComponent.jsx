import React from 'react';
import PropTypes from 'prop-types';
import FormErrorComponent from '../../../../common/FormErrorComponent/FormErrorComponent';
import styles from './EditEmailFormComponent.module.scss';

const EditEmailFormComponent = ({
  userProfile, handleEmailUpdate, emailError, setEmail, setEmailError,
}) => {
  const { email, id } = userProfile;
  const changeEmail = email => {
    setEmailError(null);
    setEmail(email);
  };

  return (
    <div className={styles.editEmailFormContainer}>
      <form onSubmit={event => handleEmailUpdate(event)}>
        <div className={styles.textInput}>
          <label htmlFor="email">
            <h3>Your Email Address</h3>
            <FormErrorComponent errorMessage={emailError} />
            <input id="email" type="email" defaultValue={email} onChange={event => changeEmail(event.target.value)} />
          </label>
        </div>
        <input type="hidden" id="id" value={id} />
        <div className={styles.editEmailSubmitButton}>
          <button type="submit">
            Change Email
          </button>
        </div>
      </form>
    </div>
  );
};

EditEmailFormComponent.defaultProps = {
  emailError: '',
  setEmailError: null,
};

EditEmailFormComponent.propTypes = {
  userProfile: PropTypes.shape({
    email: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
  }).isRequired,
  handleEmailUpdate: PropTypes.func.isRequired,
  emailError: PropTypes.string,
  setEmail: PropTypes.func.isRequired,
  setEmailError: PropTypes.func,
};

export default EditEmailFormComponent;
