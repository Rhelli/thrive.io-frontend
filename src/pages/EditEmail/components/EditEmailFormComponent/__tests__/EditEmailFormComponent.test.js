import React, { useState as useStateMock } from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import EditEmailFormComponent from '../EditEmailFormComponent';

Enzyme.configure({ adapter: new Adapter() });

const userProfile = {
  email: 'steve@email.com',
  id: 1,
};

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

const setEmail = jest.fn();
const setEmailError = jest.fn();
const emailError = null;

const handleEmailUpdate = jest.fn(() => 'Clicked!');

const editEmailFormComponent = (
  <EditEmailFormComponent
    userProfile={userProfile}
    handleEmailUpdate={handleEmailUpdate}
    emailError={emailError}
    setEmail={setEmail}
    setEmailError={setEmailError}
  />
);

describe('The appreance and functionality of the EditEmailForm component', () => {
  const container = mount(editEmailFormComponent);

  beforeEach(() => {
    useStateMock.mockImplementation(init => [init, setEmail]);
    useStateMock.mockImplementation(init => [init, setEmailError]);
  });

  it('Matches the previous snapshot', () => {
    const tree = renderer.create(editEmailFormComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has a main container and form with the correct attributes', () => {
    expect(container.childAt(0).hasClass('editEmailFormContainer')).toBe(true);
    expect(container.childAt(0).childAt(0).name()).toBe('form');
  });

  it('Will sumbit the form if it is manually submitted at form level', () => {
    container.childAt(0).childAt(0).simulate('submit');
    expect(handleEmailUpdate).toHaveBeenCalledTimes(1);
  });

  it('Displays the email address area correctly', () => {
    expect(container.find('.textInput').childAt(0).childAt(0).text()).toBe('Your Email Address');
  });

  it('Pre-populates the email input field with the users current email address', () => {
    expect(container.find('.textInput').childAt(0).childAt(2).instance().value).toBe('steve@email.com');
  });
});
