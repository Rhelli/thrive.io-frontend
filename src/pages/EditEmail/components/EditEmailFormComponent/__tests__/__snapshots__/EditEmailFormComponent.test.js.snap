// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appreance and functionality of the EditEmailForm component Matches the previous snapshot 1`] = `
<div
  className="editEmailFormContainer"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="textInput"
    >
      <label
        htmlFor="email"
      >
        <h3>
          Your Email Address
        </h3>
        <p
          className="errorMessage"
        />
        <input
          defaultValue="steve@email.com"
          id="email"
          onChange={[Function]}
          type="email"
        />
      </label>
    </div>
    <input
      id="id"
      type="hidden"
      value={1}
    />
    <div
      className="editEmailSubmitButton"
    >
      <button
        type="submit"
      >
        Change Email
      </button>
    </div>
  </form>
</div>
`;
